### DISCUSSION TITLE
Jamulus Connections history

### DISCUSSION THREAD

**Discussion by FredNor78** [jamulussoftware/jamulus#2586] [Ideas]
Hi everybody,

Sometimes we want to connect again to some servers we had visited in the past.
May be it could be a good idea to implement sort of an history of the last Directory/Server names, in order to facilitate this

What do you think of that?

Thanks

**Comment by dcorson-ticino-com** [comment]
Having a history or favorites has been discussed and implemented several times, but it has never gotten through the development discussions into a release.  Maybe it would be time to start another attempt.

**Reply by rdica** [reply]
Is this different from the history of manually typed in server addresses?

**Reply by FredNor78** [reply]
Indeed it is. As the history you mention is already kept in the drop list

**Comment by gilgongo** [comment]
I think last time we couldn't agree on a non-confusing way to show favourites without a substantial UI change. But there would be a number of possible ways of doing it so maybe this could be re-examined.

The previous discussion is here: https://github.com/jamulussoftware/jamulus/discussions/1511

**Reply by FredNor78** [reply]
Well, in fact my purpose was not about favorites but actually history

**Reply by gilgongo** [reply]
Yes, although the reason given for having it ("Sometimes we want to connect again to some servers we had visited in the past") could be met with either a history or a favorites list. 

A favourites list implies you have to nominate favorites in some way in the UI (possibly while connected to a server). A history implies a passive list sorted by date on which you connected. But a history implies you have to rely on your memory about which servers you want to connect to again, so it doesn't seem the best way of meeting the requirement of allowing you to get back to specific servers.

**Reply by FredNor78** [reply]
It probably implies that you always work with your "Favorites" and never have needed to appeal to your history.

So be it ;)

**Reply by gilgongo** [reply]
Regardless of whether we're characterising it as favourites or history though, the main question is how is that executed in the UI. 

One of the problems we have with the current Jamulus UI is that the overall structure isn't very suitable to adding new controls (eg a favourites button or a history view) unless they live in the menus and/or spawn sub windows.  We are starting to look at this in more detail though, and having a "backlog" of features like this to consider helps I think.
